python/tk_nuke_writenode/handler.py

```diff
@@ -744,6 +744,13 @@ def on_before_render_gizmo_callback(self):
         if grp:
             self.__currently_rendering_nodes.add(grp)
 
+        # Run any beforeRender code that the user added in the node's Python
+        # tab manually.
+        cmd = grp.knob("tk_before_render").value()
+
+        if cmd:
```

Comment:
It handles it reasonably gracefully, with a relevant exception raised referencing the code that was exec'd. I did go ahead and add a log_error for each to give a bit more information to the user along with the exception being raised.